File: BizBuddyProject/BizBuddyProject/Customer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BizBuddyProject.Classes
{
    public class Customer
    {
        private Guid id = Guid.NewGuid();
        private string firstName;
        private string middleName;
        private string lastName;
        private Dictionary<string, List<string>> contacts = new Dictionary<string, List<string>>();
        
        public int ID
        {
            get => this.id; 
        }

        public string FirstName
        {
            get => this.firstName; 
            set => this.firstName = value;
        }

        public string MiddleName
        {
            get => this.middleName;
            set => this.middleName = value;
        }

        public string LastName
        {
            get => this.lastName;
            set => this.lastName = value;
        }

        public int AddContact(string type, string contact)
        {
            int flag = 0;
            if(this.contacts.ContainsKey(type))
            {
                contacts[type].Add(contact);
                flag = 1;
            }
            else
            {
                contacts.Add(type, new List<string> {contact});
                flag = 1;
            }
            return flag;
        }

        public int RemoveContact(string type, string contact)
        {
            int flag = 0;
            if(this.contacts.ContainsKey(type))
            {
                contacts[type].Remove(contact);
                flag = 1;
            }
            return flag;
        }

        public int EditContact(string type, string oldContact, string newContact)
        {
            int index = contacts[type].IndexOf(oldContact);
            if(index != -1)
            {
                contacts[type][index] = newContact;
            }
            return (index != -1);
        }
    }
}

File: lib/classes/customer.dart
import 'dart:collection';

class Customer {
  String id ;
  String firstName;
  String middleName;
  String lastName;
  Map<String, List<String>> contacts = {};

  String get ID => id;

  String getFirstName() => firstName;

  void setFirstName(String value) {
    firstName = value;
  }

  String getMiddleName() => middleName;

  void setMiddleName(String value) {
    middleName = value;
  }

  String getLastName() => lastName;

  void setLastName(String value) {
    lastName = value;
  }

  int addContact(String type, String contact) {
    int flag = 0;
    if (contacts.containsKey(type)) {
      contacts[type]!.add(contact);
      flag = 1;
    } else {
      contacts[type] = [contact];
      flag = 1;
    }
    return flag;
  }

  int removeContact(String type, String contact) {
    int flag = 0;
    if (contacts.containsKey(type)) {
      contacts[type]!.remove(contact);
      flag = 1;
    }
    return flag;
  }

  int editContact(String type, String oldContact, String newContact) {
    int index = contacts[type]!.indexOf(oldContact);
    if (index != -1) {
      contacts[type]![index] = newContact;
    }
    return (index != -1) ? 1 : 0;
  }

  Customer(String firstName, String middleName, String lastName)
  {
    this.id = idGenerator();
    this.firstName = firstName;
    this.middleName = middleName;
    this.lastName = lastName;
  }

  String idGenerator() {
    final now = DateTime.now();
    return now.microsecondsSinceEpoch.toString();
  }
}